hooks/publish.py

```diff
@@ -305,24 +305,13 @@ def validate(self, task_settings, item):
 
         # ---- validate the settings required to publish
 
-        attr_list = ("publish_type", "publish_version", "publish_name", "publish_path",
-                     "publish_symlink_path", "publish_linked_entity_name")
-        for attr in attr_list:
-            try:
-                method = getattr(self, "_get_%s" % attr)
-                item.properties[attr] = method(task_settings, item)
-            except Exception:
-                self.logger.error(
-                    "Unable to determine '%s' for item: %s" % (attr, item.name),
-                    extra={
-                        "action_show_more_info": {
-                            "label": "Show Error Log",
-                            "tooltip": "Show the error log",
-                            "text": traceback.format_exc()
-                        }
-                    }
-                )
-                return False
+        global_attr_list = ("publish_version", "publish_linked_entity_name")
```

Comment:
These should local too.

Reply:
When we are publishing an item, all its plugins should publish the same version. Also, all should be linked to the same entity (this is asset, shot etc.)  I can't think of a scenario where they are plugin specific and so I left them in the global properties for the item.

Reply:
Honestly, in practice this really doesn't matter as nothing else in our setup really relies on either of these except for subclassed plugins. In principal, I'm not a huge fan of defining implicit dependencies between plugins, especially since we have no explicit mechanism for defining precedence, blocking, and data dependencies between them besides the order in which they are defined in the config.

Take, for example, the situation where a user decides to disable the "Publish Session" plugin in the UI but not the "Create Version" plugin. Now `item.properties.publish_version` doesn't get initialized because `validate()` doesn't get run for "Publish Session", and consequently "Create Version" errors out because its relying on that value. I'd personally like to see any/all global properties defined at collection when we can ensure they will be properly initialized, but I know that isn't always practical / desirable. Alternatively, we need to ensure we are adding better error handling in our publish plugins for when those shared properties aren't defined. In the case of "Create Version", that could mean updating it to inherit/run the `_get_publish_version()` method if `publish_version` isn't already defined.